research/fed-sm/README.md

```diff
@@ -0,0 +1,82 @@
+It uses [MONAI](https://github.com/Project-MONAI/MONAI),
+which is a PyTorch-based, open-source framework for deep learning in healthcare imaging, part of the PyTorch Ecosystem.
+
+### FedSM
+This example illustrates the personalized federated learning algorithm [FedSM](https://arxiv.org/abs/2203.10144) accepted to [CVPR2022](https://cvpr2022.thecvf.com/). 
+It bridges the different data distributions across clients via a SoftPull mechanism and utilizes a Super Model. 
+
+## Multi-source Prostate Segmentation
+This example uses 2D (axial slices) segmentation of the prostate in T2-weighted MRIs based on multiple datasets.
+
+Please refer to [Prostate Example](https://github.com/NVIDIA/NVFlare/tree/dev/examples/advanced/prostate) for details of data preparation and task specs. In the following, we assume the data has been prepared in the same way to `${PWD}/data_preparation`. The dataset is saved to `${PWD}/data_preparation/dataset_2D`, and datalists are saved to `${PWD}/data_preparation/datalist_2D`.
+
+## Run automated experiments
+We use the NVFlare simulator to run FL training automatically, the 3 clients are named `client_I2CVB, client_MSD, client_NCI_ISBI_3T`
+### Prepare local configs
+First, we add the image directory root to `config_train.json` files for generating the absolute path to dataset and datalist. 
+In the current folder structure, it will be `${PWD}/..`. 
+It can be any arbitrary path where the data locates.  
+```
+for job in fedsm_prostate
+do
+  sed -i "s|DATASET_ROOT|${PWD}/data_preparation|g" jobs/${job}/app/config/config_train.json
+done
+```
+### Use NVFlare simulator to run the experiments
+We use NVFlare simulator to run the FL training experiments, following the pattern:
+```
+nvflare simulator jobs/[job] -w ${workspace_path}/[job] -c [clients] -gpu [gpu] -t [thread]
+```
+`[job]` is the experiment job that will be submitted for the FL training. 
+In this example, this is `fedsm_prostate`.  
+The combination of `-c` and `-gpu`/`-t` controls the resource allocation. 
+
+## Results on three clients for FedSM
+In this example, we run three clients on 1 GPU with three threads `-t 3`. The minimum GPU memory requirement is 12 GB. We put the workspace in `/tmp` folder
+```
+nvflare simulator jobs/fedsm_prostate -w /tmp/nvflare/fedsm_prostate -c client_I2CVB,client_MSD,client_NCI_ISBI_3T -t 3
+```
+
+### Validation curve on each site
+In this example, each client computes their validation scores using their own
+validation set. 
+
+We provide a script for plotting the TensorBoard records:
+```
+python3 ./result_stat/plot_tensorboard_events.py
+```
+The TensorBoard curves (smoothed with weight 0.8) for validation Dice for the 100 epochs (100 rounds, 1 local epoch per round) during training are shown below:
+![All training curve](./figs/all_training.png)
+
+As shown, FedSM personalized models can achieve better validation accuracy than global model, and the selector model is able to distinct among sites  
```

Comment:
Misses full-stop. Maybe add a comment on the stability of model selector training?

Reply:
added a paragraph in FedSM section